client/gui/CAnimation.cpp

```diff
@@ -753,6 +779,24 @@ void SDLImage::shiftPalette(int from, int howMany)
 	}
 }
 
+void SDLImage::adjustPalette(ColorShifter * shifter)
+{
+	SDL_Palette* palette = surf->format->palette;
+	for (int i = 0; i < 255; i++)
+	{
+		palette->colors[i] = shifter->shiftColor(originalPalette->colors[i]);

```

Comment:
You should not alter whole palette. There are a few special color indexes that should not be changed (see CDefFile::CDefFile for reference).

Reply:
I will update soon.